feat: switch to xmake (#377)
UE4SS-RE/RE-UE4SS: README.md

    Make sure your Github account is linked to your Epic Games account for UE source access.
    Do not use the `--remote` option because that will force third-party dependencies to update to the latest commit, and that can break things.
    You will need your github account to be linked to an Epic games account to pull the Unreal pseudo code submodule.

> There are three different ways you can build UE4SS.

Reviewer: This says there are three different ways, but there's only two methods below.

* `Target`
  * `Game` - for regular games
  * `CasePreserving` - for games built with case preserving enabled

* `Config`
  * `Dev` - development build
  * `Debug` - debug build
  * `Shipping` - shipping(release) build
  * `Test` - build for tests

* `Platform`
  * `Win64` - 64-bit windows

Now to build it, just run `xmake`

### Opening in an IDE

#### Visual Studio / Rider

To generate Visual Studio project files, run the `xmake project -k vsxmake2022` command.

Afterwards open the generated `.sln` file inside of the `vsxmake2022` directory

Note that you should also commit & push the submodules that you've updated if the reason why you updated was not because someone else pushed an update, and you're just catching up to it.

## Updating git submodules

If you want to update git submodules, you do so one of three ways:
1. You can execute `remote_update_first_party_submodules.bat` to update all first-party dependencies.
2. You can also choose to update dependencies one by one, by executing `git submodule update --init --recursive vendor/<RepoOwner>/<Repo>`.
    Remember to not use the `--remote` option unless you actually want to update to the latest commit.

> 3. If you would rather pick a specific commit or branch to update a dependency to then `cd` into the submodule directory for that dependency and execute `git checkout <branch name or commit>`.

Reviewer: The file `remote_update_first_party_submodules.bat` is an extremely old script that likely hasn't been updated in a long time, and in fact, it was removed from this repo some time ago.
It does still exist in the Unreal repo but as I said, it's extremely old so it shouldn't be used and won't work if it is used, in fact, it should be removed.

This section should be adjusted to not mention this file.
It's so old that it refers to a non-existing `vendor` directory.

> ## Updating git submodules
> 
> If you want to update git submodules, you do so one of three ways:
> 1. You can execute `git submodule update --init --recursive` to update all first-party submodules.
> 2. You can also choose to update submodules one by one, by executing `git submodule update --init --recursive vendor/<RepoOwner>/<Repo>`.
>     Remember to not use the `--remote` option unless you actually want to update to the latest commit.
> 3. If you would rather pick a specific commit or branch to update a submodule to then `cd` into the submodule directory for that submodule and execute `git checkout <branch name or commit>`.
> 

Reviewer: ```suggestion
## Updating git submodules

If you want to update git submodules, you do so one of three ways:
1. You can execute `git submodule update --init --recursive` to update all submodules.
2. You can also choose to update submodules one by one, by executing `git submodule update --init --recursive deps/<first-or-third>/<Repo>`.
    Do not use the `--remote` option unless you actually want to update to the latest commit.
3. If you would rather pick a specific commit or branch to update a submodule to then `cd` into the submodule directory for that dependency and execute `git checkout <branch name or commit>`.
The main dependency you might want to update from time to time is `deps/first/Unreal`.

```
